packages/web-components/src/tooltip/tooltip.ts

```diff
@@ -0,0 +1,198 @@
+import { attr, FASTElement, nullableNumberConverter } from '@microsoft/fast-element';
+import { uniqueId } from '@microsoft/fast-web-utilities';
+
+const SUPPORTS_ANCHOR_POSITIONING = CSS.supports('anchor-name: --a');
+
+/**
+ * A Tooltip Custom HTML Element.
+ * Based on ARIA APG Tooltip Pattern {@link https://www.w3.org/WAI/ARIA/apg/patterns/tooltip/ }
+ * @public
+ */
+export class Tooltip extends FASTElement {
+  /**
+   * The attached element internals
+   */
+  public elementInternals = this.attachInternals();
+
+  /**
+   * The item ID
+   *
+   * @public
+   * @remarks
+   * HTML Attribute: id
+   */
+  @attr
+  public id: string = uniqueId('tooltip-');
+
+  /**
+   * Set the delay for the tooltip
+   */
+  @attr({ converter: nullableNumberConverter })
+  public delay?: number;
+
+  /**
+   * The default delay for the tooltip
+   * @internal
+   */
+  private defaultDelay: number = 250;
+
+  /**
+   * Set the positioning of the tooltip
+   */
+  @attr
+  public positioning?: string;
+
+  /**
+   * The id of the anchor element for the tooltip
+   */
+  @attr
+  public anchor: string = '';
+
+  /**
+   * Reference to the anchor element
+   * @internal
+   */
+  private get anchorElement(): HTMLElement | null {
+    const rootNode = this.getRootNode();
+    return (rootNode instanceof ShadowRoot ? rootNode : document).getElementById(this.anchor ?? '');
+  }
+
+  /**
+   * Reference to the anchor positioning style element
+   * @internal
+   */
+  protected anchorPositioningStyleElement: HTMLStyleElement | null = null;
+
+  public constructor() {
+    super();
+    this.elementInternals.role = 'tooltip';
+  }
+
+  public connectedCallback(): void {
+    super.connectedCallback();
+    if (this.anchorElement) {
+      // @ts-expect-error - Baseline 2024
+      const anchorName = this.anchorElement.style.anchorName || `--${this.anchor}`;
+      this.anchorElement.setAttribute('aria-describedby', this.id);
```

Comment:
Since `aria-describedby` accepts multiple space-separated IDREFs, maybe we need to add a bit more code to cover that case, like:

```ts
const describedBy = this.anchorElement.getAttribute('aria-describedby');
this.anchorElement.setAttribute(
  'aria-describedby',
  describedBy ? `${describedBy} ${this.id}` : this.id
);
```